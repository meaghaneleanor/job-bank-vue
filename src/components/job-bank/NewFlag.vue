<template>
  <div v-if="checkTimeDifference(date)"
       class="ontario-grid-cell large-1 large-offset-11 small-2 small-offset-10">
    <p>{{ $t('jobPage.newFlag') }}</p>
  </div>
</template>

<script>
export default {
  props: {
    date: {
      type: String,
      required: true
    }
  },
  methods: {
    checkTimeDifference(date) {
      //get todays date in milliseconds
      let todaysDate = Date.now();

      //get job date in milliseconds
      let parsedDate = Date.parse(date);

      //compare time difference in milliseconds
      let timeDifference = todaysDate - parsedDate;
      
      let duration = 60000 * 60 * 24;
      let dayDifference = Math.round(timeDifference/duration);

      return dayDifference > process.env.VUE_APP_JB_NEW_FLAG ? false: true;
    }
  }
};
</script>
