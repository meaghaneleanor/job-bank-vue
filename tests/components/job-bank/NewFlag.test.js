import { shallowMount } from '@vue/test-utils';
import { $t, i18nMock } from '../../test.utils.js';

import NewFlag from '../../../src/components/job-bank/NewFlag.vue';

const defaultOptions = {
  mocks: {
    $t, 
    i18nMock,
    
  },
  propsData:{
    date: '2019-07-10T10:59:00Z'
  }
};

const createComponent = (options = defaultOptions) => shallowMount(NewFlag, options);

describe('NewFlag.vue', () => {
  let wrapper;
  
  describe('snapshot', () => {
    it('should display the expected html', () => {
      wrapper = createComponent();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('.checkTimeDifference', () => {
      describe('if the job was posted more than 3 days ago', () => {
        let date = '2019-07-10T10:59:00Z';

        it('should return false', () => {
          expect(wrapper.vm.checkTimeDifference(date)).toEqual(false);
        });
      });

      describe('if the job was posted within 3 days from the current date', () => {
        it('should return false', () => {
          expect(wrapper.vm.checkTimeDifference(Date.now())).toEqual(true);
        });
      });
    });
  });
});
