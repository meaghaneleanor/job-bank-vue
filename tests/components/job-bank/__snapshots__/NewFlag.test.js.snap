// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewFlag.vue snapshot should display the expected html 1`] = `"<div class=\\"ontario-grid-cell large-1 large-offset-11 small-2 small-offset-10\\"><p>jobPage.newFlag</p></div>"`;
